### ISSUE TITLE
p-calendar styling is incorrect when it is disabled with an icon

### ISSUE THREAD

**Issue by PeterHewat** [primefaces/primeng#10167]
#10024 was only partially fixed then closed hence I am opening a new issue to bring forward the remaining issue.

**I'm submitting a ...**
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Stackblitz Example**
Previous bug #10024 in PrimeNG 11.3.0:
https://stackblitz.com/edit/primeng-issue-mdc-theme-calendar-with-icon-primeng-11-3-0?file=src%2Fapp%2Fapp.component.html

Partial fix in PrimeNG 11.4.0
https://stackblitz.com/edit/primeng-issue-mdc-theme-calendar-with-icon-primeng-11-4-0?file=src%2Fapp%2Fapp.component.html

**Current behavior**
The style of a disabled p-calendar with icon is incorrect
=> hovering over it changes it's style.
This doesn't occur if the calendar icon isn't displayed.

**Expected behavior**
The style of a disabled p-calendar should be correct, irrelevant of if it has an icon or not.

**Minimal reproduction of the problem with instructions**
See Stackblitz Example
*Prerequisite:* Use class p-input-filled & themes/mdc-light-indigo/theme.css.

* **Angular version:** 11.2.11

* **PrimeNG version:** 11.4.0

* **Browser:** all
 
* **Language:** all

**Comment by PeterHewat** [comment]
Adding styleClass="p-disabled" to the disabled p-calendar as described in #10024 fixes this issue.
